openeo/internal/jupyter.py

```diff
@@ -102,6 +102,8 @@ def render_component(component: str, data = None, parameters: dict = None):
     # Set the data as the corresponding parameter in the Vue components
     key = COMPONENT_MAP.get(component, component)
     if data is not None:
+        if isinstance(data, list):
+            data = [(x.to_dict() if "to_dict" in dir(x) else x) for x in data]
```

Comment:
it's more idiomatic (and more efficient I guess too) to do `if hasattr(x, "to_dict")`  than `if "to_dict" in dir(x)`

Reply:
```suggestion
            data = [(x.to_dict() if hasattr(x, "to_dict") else x) for x in data]
```